### PR TITLE
Roc 93 layout chat service pattern

### PR DESCRIPTION
**Pull request by LarrySchultheis**
Creating draft PR for chat stuff to link in doc. So far have basic socket communication established.

### CODE REVIEW

File: rockr/views.py [hunk 1]
```diff
@@ -97,3 +109,24 @@ def get_user_role():
 def get_roles():
     resp = uq.get_roles()
     return format_response(resp["status"], resp["data"])
+
+@app.route('/matches', methods=["GET"])
+def get_matches():
+    user = User.query.filter_by(email=request.args.get("email")).first();
+    matches = db.session.query(User, UserMatch).join(User, User.id == UserMatch.user_id).filter(UserMatch.match_id == user.id).all()
```
Comment (squaill) [inline rockr/views.py:115]: image

Comment (LarrySchultheis) [reply rockr/views.py:115]: The fact that you went through the time to make that meme says a lot. Will refactor lmao

File: tests/test_rockr.py [hunk 2]
```diff
@@ -189,7 +189,8 @@ def test_goals(self):
 
     def test_matches(self):
         user = User.query.filter_by(email=TEST_EMAIL).first();
-        matches = db.session.query(User, UserMatch).join(User, User.id == UserMatch.user_id).filter(UserMatch.match_id == user.id).all()
+        matches = UserMatch.query(User, UserMatch).join(User, User.id == UserMatch.user_id).filter(UserMatch.match_id == user.id).all()
```
Comment (squaill) [inline tests/test_rockr.py]: are you trying to get the matches for `user`? If so, you can just do `UserMatch.query.filter_by(user_id=user.id)`

File: tests/test_rockr.py [hunk 3]
```diff
@@ -189,7 +189,8 @@ def test_goals(self):
...
+        assert matches is not None
         assert len(matches) > 0
```
Comment (squaill) [inline tests/test_rockr.py]: if you do the above, this will probably need to be `matches.count()`

File: tests/test_rockr.py (hunk 2 above)
Comment (LarrySchultheis) [reply tests/test_rockr.py]: Need join on user to get data contained within that model such as name

Comment (squaill) [reply tests/test_rockr.py]: what do you need name for here?

Comment (LarrySchultheis) [reply tests/test_rockr.py]: To display the matched user names in the chat page

Comment (squaill) [reply tests/test_rockr.py]: isn't this test just to check that matches exist?

Comment (LarrySchultheis) [reply tests/test_rockr.py]: Nah it is to get the matches for the user so the chat page can display their chats. Just reflecting the logic that exists in the endpoint until I get a chance to refactor it.

Comment (LarrySchultheis) [reply tests/test_rockr.py]: This PR is also currently draft :)

File: rockr/views.py [hunk 4]
```diff
@@ -44,12 +58,78 @@ def get_user_role():
     }
     return jsonify(data)
 
+@app.route('/update_user_account', methods=["POST"])
+def update_user_account():
```
Comment (squaill) [inline rockr/views.py:66]: your endpoints shouldn't have the request verb in them. We can be more generic. Let's touch base on this tomorrow

**Review by squaill — CHANGES_REQUESTED** [review]

File: rockr/utils/message_handler.py [hunk 5]
```diff
@@ -0,0 +1,12 @@
+from rockr import app, db, db_manager
+from rockr.models import Message, User
+from datetime import datetime
+
+def save_message(req):
+    print(req['sender']['email'])
```
Comment (squaill) [inline rockr/utils/message_handler.py:6]: bad Laer Bear

**Comment by LarrySchultheis** [comment]
Imma be super dooper real with you rn. I don't care about this project anymore

**Comment by LarrySchultheis** [comment]
Better to ask for forgiveness than ask for permission

**Comment by squaill** [comment]
more work for me later but who cares

**Comment by LarrySchultheis** [comment]
I vaguely remember us agreeing to not step on each other's toes. I ain't been doin any toe steppin ;)

**Pull request merged**